Bubble Sort Algorithm

smcommits opened the pull request:
**Project Description** 

In this project we have implemented _bubble sort_ and _bubble sort by_ algorithms in the Ruby programming language. _Bubble Sort_ is a simple sorting algorithm that repeatedly steps through the list, compares adjacent elements and swaps them if they are in the wrong order. The pass through the list is repeated until the list is sorted.  _Bubble Sort By_ algorithm is smilar to the bubble sort except the items are sorted based on the length of elements

**Key Points**

The algorithms are optimized based on two aspects of the bubble sort, first is the running the algorithm only until the list is swapped. We have achieved this using a boolean vairable for called _'switched'_ and second is reducing the number of elements on each iteration. We have achieved this by storing the length of the array in variabl _n_ and reducing the n by 1 at each iteration.

smcommits committed:
Implement bubble sort algorithm

Initialized a variable for array length
Initialized a boolean variable for if elements switch
Compared the two elements using swapping variables

Misori-simon committed:
Implemented the bubble sort by method
-initialised the input array length
-set the switched variable to false
-performed block consitions on yielded variables

smcommits committed:
Change name workflow folder

Fixed the name error in github/workflow folder
Changed it from .workflow to workflow

Misori-simon committed:
Change folder name
-cahnge name of workflows folder

smcommits committed:
Add Linter Config and Fix readme

Added the github and workflows folder
Added rubocop.yml for linting configuration
Fixed the incorrect readme file

Misori-simon committed:
Update Readme

-add project info and description
-authors info

smcommits committed:
Update readme

Added the image for bubble sort
Added the codeblocks in README.md

umairarshadbutt reviewed (changes requested):
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @Misori-simon and @smcommits
Such a remarkable project you made :+1: ! Awesome README :100: !
We, however, do have a few changes to carry out before we proceed (see review comment).

##

 We only need to implement 1 fix. Kindly check the comments below  👇:
- [x] Let us create a separate file for bubble_sort.rb and bubble_sort_by.rb
- [x] Add the MIT License in the README file. (MIT License is missing in the repository)
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

smcommits committed:
Implement change request

Fixed changes requested in pull-request
 - Divided the bubble_sort and bubble_sort_by into two different files

Misori-simon committed:
Update readme

-removed MIT license

umairarshadbutt reviewed (approved):
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##

**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @Misori-simon and @smcommits! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal: 

- **[OPTIONAL]:** You can add MIT License.
##
**If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
